src/request.ts

```diff
@@ -19,11 +19,17 @@ const getHeaders = (requestProperties: RequestInit): HeadersInit => {
     const headers = requestProperties.headers || {};
     if ('body' in requestProperties) {
         if (headers instanceof Headers) {
-            headers.set('Content-Type', 'application/json');
+            if (!headers.has('Content-Type')) {
+                headers.set('Content-Type', 'application/json');
+            }
         } else if (Array.isArray(headers)) {
-            headers.push(['Content-Type', 'application/json']);
+            if (headers.reduce((previous, value) => previous && value[0] !== 'Content-Type', true)) {
```

Comment:
```suggestion
            if (!headers.some(header => header[0] === 'Content-Type')) {
```
Seems more readable and less confusing to me